server/index.js

```diff
@@ -56,6 +56,105 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.json([]);
+  } else {
+    const sql = `
+      select "c"."cartItemId",
+             "c"."price",
+             "p"."productId",
+             "p"."image",
+             "p"."name",
+             "p"."shortDescription"
+        from "cartItems" as "c"
+        join "products" as "p" using ("productId")
+      where "c"."cartId" = $1
+    `;
+    const params = [req.session.cartId];
+    db.query(sql, params)
+      .then(result => {
+        const cartItems = result.rows;
+        res.status(200).json(cartItems);
+      })
+      .catch(err => next(err));
+  }
+});
+
+app.post('/api/cart/', (req, res, next) => {
+  const productId = Number(req.body.productId);
+  if (!Number.isInteger(productId) || productId <= 0) {
+    next(new ClientError('"productId" must be a positive integer', 400));
+  }
+  const sql = `
+    select "price"
+      from "products"
+     where "productId" = $1
+  `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const price = result.rows[0].price;
+      if (!price) {
```

Comment:
The price could _technically_ be `0`, which is falsy. Instead, you can base your conditional expression on the `length` of the rows array or the `result.rowCount`.

Reply:
<img width="872" alt="Screen Shot 2020-06-14 at 10 18 23 AM" src="https://user-images.githubusercontent.com/60276373/84599711-f9238180-ae28-11ea-84ba-8c10937e8885.png">